### PR TITLE
[Merged by Bors] - Add model MobileNetV1 and MobileNetV2

### LINKED ISSUE #5: Support more model examples
MobileNet V1, V2.

### PR DESCRIPTION
**Pull request by ChengcanWang-zte**
Closes #5.

Signed-off-by: Chengcan Wang <wang.chengcan@zte.com.cn>

### CODE REVIEW

**Comment by ChengcanWang-com** [comment]
Close #5

File: examples/mobilenet_v1_imagenet_train.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# sys.path.insert(0, join(abspath(dirname(__file__)), '../src'))
+# print(sys.path)
+
+from model_optimizer import prune_model  # noqa: E402
+
+
+def _main():
+    base_dir = os.path.dirname(__file__)
+    request = {
+        "dataset": "imagenet",  # or imagenet
```
Comment (EFanZh) [inline examples/mobilenet_v1_imagenet_train.py]: Invalid comment.

File: src/model_optimizer/pruner/models/mobilenet_v1.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    :param num_classes:
+    :param dropout_prob:
+    :param is_training:
+    :param depth_multiplier:
+    :return:
+    """
+    return _mobilenet_v1(num_classes, dropout_prob, is_training, scale=1.0, depth_multiplier=depth_multiplier)
+
+
+def _mobilenet_v1(num_classes=1000,
```
Comment (EFanZh) [inline src/model_optimizer/pruner/models/mobilenet_v1.py:86]: Why not use `keras.applications.MobileNet` directly?

**Review by EFanZh — CHANGES_REQUESTED** [review]
Can those “.gitkeep” file be remove and have those directories being created on-demand?

File: examples/mobilenet_v1_imagenet_train.py (hunk 1 above)
Comment (ChengcanWang-zte) [reply examples/mobilenet_v1_imagenet_train.py]: ok. Delete already created directories, these directories will be created automatically by the program.

File: src/model_optimizer/pruner/models/mobilenet_v1.py (hunk 2 above)
Comment (ChengcanWang-zte) [reply src/model_optimizer/pruner/models/mobilenet_v1.py:86]: There are several reasons:
1. Keras.applications.MobileNet does not define initialze regularity and other parameters.
2. The BN layer of the keras model does not pass in the is_training parameter. Dropout is also the best infer and training to distinguish whether there is dropout.
3. The keras.applications.MobileNet pre-training model is actually transferred from the tensorflow model, why is it not trained by keras itself. It's a bit strange.

File: src/model_optimizer/__init__.py [hunk 3]
```diff
@@ -3,13 +3,18 @@
...
+import os
 from .quantizer import create_quantizer
 from .quantizer.config import create_config_from_obj as quant_conf_from_obj
 from .pruner.config import create_config_from_obj as prune_conf_from_obj
 from .pruner.runner import run_scheduler
 
 
+def _make_dirs(file_path):
+    if not os.path.exists(file_path):
+        os.makedirs(file_path)
```
Comment (EFanZh) [inline src/model_optimizer/__init__.py]: Use `os.makedirs(file_path, exists_ok=True)`.

**Review by EFanZh — CHANGES_REQUESTED** [review]

**Comment by EFanZh** [comment]
@ChengcanWang-com The CI tests have failed.
